### ISSUE TITLE
Coinbase Advanced Trade - Failed to submit order due to KeyError: 'order_id'

### ISSUE THREAD

**Issue by rapcmia** [hummingbot/hummingbot#7211]
### Describe the bug

When checking coinbase advanced trade, orders are failing on hummingbot but successfully posted on exchange

![image](https://github.com/user-attachments/assets/aa5dc461-4a3e-4de6-8a02-299ccd7818fc)
```
2024-09-09 16:06:25,401 - 19328 - hummingbot.client.hummingbot_application - INFO - start command initiated.
2024-09-09 16:06:26,096 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_exchange.CoinbaseAdvancedTradeExchange - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2024-09-09 16:06:28,053 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_api_user_stream_data_source.CoinbaseAdvancedTradeAPIUserStreamDataSource - INFO - Subscribe-ing to user for ['BTC-USDC'] ...
2024-09-09 16:06:28,092 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_api_order_book_data_source.CoinbaseAdvancedTradeAPIOrderBookDataSource - INFO - Subscribed to order book channels for: BTC-USDC
2024-09-09 16:06:28,374 - 19328 - hummingbot.core.data_type.order_book_tracker - INFO - Initialized order book for BTC-USDC. 1/1 completed.
2024-09-09 16:06:30,003 - 19328 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (BTC-USDC) Creating 1 bid orders at (Size, Price): ['0.0001 BTC, 54562.01 USDC']
2024-09-09 16:06:30,005 - 19328 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (BTC-USDC) Creating 1 ask orders at (Size, Price): ['0.0001 BTC, 55664.27 USDC']
2024-09-09 16:06:30,201 - 19328 - hummingbot.core.rate_oracle.rate_oracle - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2024-09-09 16:06:30,381 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_exchange.CoinbaseAdvancedTradeExchange - NETWORK - Error submitting buy LIMIT_MAKER order to Coinbase_advanced_trade for 0.00010000 BTC-USDC 54562.01.
Traceback (most recent call last):
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange/coinbase_advanced_trade/coinbase_advanced_trade_exchange.py", line 393, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange/coinbase_advanced_trade/coinbase_advanced_trade_exchange.py", line 373, in _place_order
    o_id = str(order_result["order_id"])
KeyError: 'order_id'
2024-09-09 16:06:30,385 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_exchange.CoinbaseAdvancedTradeExchange - WARNING - Failed to submit BUY order to Coinbase_advanced_trade. Check API key and network connection.
2024-09-09 16:06:30,389 - 19328 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1725869190.0, "order_id": "CBAT-BBCUC621ab3bffc11bfbf1c33ff", "order_type": "OrderType.LIMIT_MAKER", "event_name": "MarketOrderFailureEvent", "event_source": "coinbase_advanced_trade"}
2024-09-09 16:06:30,389 - 19328 - hummingbot.connector.client_order_tracker - INFO - Order CBAT-BBCUC621ab3bffc11bfbf1c33ff has failed. Order Update: OrderUpdate(trading_pair='BTC-USDC', update_timestamp=1725869190.0, new_state=<OrderState.FAILED: 6>, client_order_id='CBAT-BBCUC621ab3bffc11bfbf1c33ff', exchange_order_id=None, misc_updates=None)
2024-09-09 16:06:30,390 - 19328 - hummingbot.connector.exchange.coinbase_advanced_trade.coinbase_advanced_trade_exchange.CoinbaseAdvancedTradeExchange - NETWORK - Error submitting sell LIMIT_MAKER order to Coinbase_advanced_trade for 0.00010000 BTC-USDC 55664.27.
Traceback (most recent call last):
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange/coinbase_advanced_trade/coinbase_advanced_trade_exchange.py", line 393, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/Users/rapcomia/github/hummingbot/hummingbot/latest/hummingbot/connector/exchange/coinbase_advanced_trade/coinbase_advanced_trade_exchange.py", line 373, in _place_order
    o_id = str(order_result["order_id"])
KeyError: 'order_id'
```
- Observed orders are posted on exchange
  ![image](https://github.com/user-attachments/assets/7bb99b52-2b94-4394-a103-f33e52dadc94)
  ![image](https://github.com/user-attachments/assets/670213da-febe-4a13-a488-63a42491718b)
  

### Steps to reproduce

1.  Setup a v1 or v2 strategy using `coinbase_advance_trade`
2. Start strategy/script and observe placing orders on hummingbot and exchange

### Release version

latest 2.0.1

### Type of installation

Source

### Attach required files

[logs_conf_pure_mm_1.log](https://github.com/user-attachments/files/16927641/logs_conf_pure_mm_1.log)

**Comment by nikspz** [comment]
Acceptance:
Update coinbase_advanced_trade API with latest changes 
https://docs.cdp.coinbase.com/advanced-trade/docs/changelog/

Bounty: 
Sponsor: Funded by Hummingbot Foundation [connector pots](https://docs.google.com/spreadsheets/d/1UNAumPMnXfsghAAXrfKkPGRH9QlC8k7Cu1FGQVL1t0M/edit?gid=2065823371#gid=2065823371)
Bounty Amount: 50K HBOT
Developer Payment: 50K HBOT

**Comment by isreallee82** [comment]
can i please work on this @nikspz

**Comment by jaykayudo** [comment]
Please, i will like to be assigned this @nikspz

**Comment by nikspz** [comment]
> can i please work on this @nikspz

Assigned to @isreallee82

**Comment by nikspz** [comment]
Assigned to @isreallee82, please be informed:
- Since you’re assigned to this bounty exclusively, please keep us informed on your progress over the next few days
- Please review and follow contribution guidelines (https://hummingbot.org/developers/contributions/)
- Please review the checklist after PR submitted (https://hummingbot.org/developers/contributions/#checklist)
- Let us know if you don’t have time, and we will unassigned and keep it open for other devs
- Only one dev can be assigned to one bounty

Thanks for your contribution!

**Comment by isreallee82** [comment]
thanks Should have a pr ready asap cc @nikspz

**Comment by isreallee82** [comment]
pr #7220

**Comment by nikspz** [comment]
fix merged to latest development (dev-2.0.2)

**Comment by cardosofede** [comment]
we need to close this one? @rapcmia @nikspz
